R/ggplotMDS.R

```diff
@@ -229,53 +343,45 @@ ggplotMDS <- function(DGEdata,
                                               varAnnot                = var.data,
                                               decorations             = decorations,
                                               graphType               = "Scatter2D",
-                                              colors                  = colors,
                                               colorBy                 = colorCol,
                                               shapeBy                 = shapeCol,
                                               sizeBy                  = sizeCol,
+                                              dataPointSize           = symSize,
                                               showDecorations         = TRUE,
                                               shapes                  = shapes,
+                                              colors                  = colors,
                                               title                   = title,
                                               xAxisTitle              = xylab[[1]],
                                               yAxisTitle              = xylab[[2]],
-                                              citation                = alabel,
+                                              citation                = citation,
                                               citationScaleFontFactor = 0.8,
                                               events                  = events)
-
     } else {
-        # Shapes: solid circle, square, triangle, diamond, open circle, square, triangle, diamond
-        ggShapes = c(16, 15, 17, 18, 21, 22, 24, 23)
-        if (missing(shapes)) {
-            shapes <- ggShapes
-        }
 
-        if (byShape == FALSE & bySize == FALSE) {
-            mdsplot <- ggplot(xydat, aes(x = x, y = y, color = ColorCode)) +
-                geom_point(shape = symShape, size = symSize, alpha = alpha)
-        } else if (byShape == TRUE & bySize == FALSE) {
-            mdsplot <- ggplot(xydat, aes(x = x, y = y, color = ColorCode, shape = Shape)) +
-                geom_point(size = symSize, alpha = alpha) +
+        shapes <- .get_valid_symbolShapes_ggplot()[1:8]
+
+        if (byShape == FALSE && bySize == FALSE) {
```

Comment:
for boolean fields you can use it directly in the comparisons or inside if statements without explicit call to **== TRUE/FALSE**